test_provider.py

```diff
@@ -0,0 +1,149 @@
+import json
+from datetime import datetime, timedelta
+from common.api import ProgramProvider
+from common.minimodel import *
+
+
+class JsonProvider(ProgramProvider):
+    def __init__(self, path):
+        self.path = path
+
+    def load_program(self, path: str) -> dict:
+        with open(path, 'r') as f:
+            return json.loads(f.read())
+    
+    @staticmethod
+    def parse_magnitude(json: dict) -> Magnitude:
+        ...
+
+    def parse_timing_window(json: dict) -> TimingWindow:
+        tw_arr = []
+        for timing_window in json['timingWindows']:
+            tw = TimingWindow()
+            tw.start = datetime.fromtimestamp(timing_window['start']/1000.0)
+            tw.duration = timedelta(milliseconds=timing_window['duration'])
+            tw.repeat = timing_window['repeat']
+            tw.period = timedelta(milliseconds=timing_window['repeat']) if timing_window['repeat'] > 0 else None 
+            tw_arr.append(tw)
+        return tw_arr
+    
+    @staticmethod
+    def parse_constraints(json: dict) -> Constraints:
+        constraints = Constraints()
+        
+        constraints.cc = json['cc']
+        constraints.iq = json['iq']
+        constraints.sb = json['sb']
+        constraints.wv = json['wv']
+        constraints.elevation_type = json['elevationConstraintType']
+        constraints.elevation_min = json['elevationConstraintMin']
+        constraints.elevation_max = json['elevationConstraintMax']
+        constraints.timing_windows = JsonProvider.parse_timing_window(json['timingWindows']) # this would need parsing 
+        constraints.strehl = ...
+    
+    @staticmethod
+    def parse_atom(json: dict) -> Observation:
+        atom = Atom()
+
+        atom.id = ...
+        atom.exec_time = ...
+        atom.prog_time = ...
+        atom.part_time = ...
+        atom.observed = json['metadata:complete']
+        atom.qa_state = ...
+        atom.guide_state = ...
+        atom.required_resources = ...
+        atom.wavelength = ...
```

Comment:
While `Atom` is not `frozen` because we will be likely adjusting the times, I think it would still be good to parse out:
```
id = ...
exec_time = ...
...
return Atom(id, exec_time, ...)
```